src/dbnode/persist/fs/msgpack/decoder_test.go

```diff
@@ -128,12 +129,19 @@ func TestDecodeIndexEntryMoreFieldsThanExpected(t *testing.T) {
 	// Intentionally bump number of fields for the index entry object
 	enc.encodeNumObjectFieldsForFn = testGenEncodeNumObjectFieldsForFn(enc, indexEntryType, 1)
 	require.NoError(t, enc.EncodeIndexEntry(testIndexEntry))
+
+	// This hokey bit of logic is done so we can add extra fields in the correct location (since new IndexEntry fields
+	// will be added *before* the checksum). Strip off checksum, add unexpected field,
+	// and re-add updated checksum value
+	enc.buf.Truncate(len(enc.Bytes()) - 5) // 5 bytes = 1 byte for integer code + 4 bytes for checksum
```

Comment:
nit: Is it worth checking the last 5 bytes for expected checksum?

Reply:
Good call! Will add that